### ISSUE TITLE
Are there any prospects to make rips_layer and perslay use  GPU  and not just CPU? thanks. 

### ISSUE THREAD

**Issue by fbarfi** [GUDHI/gudhi-deploy#29]

**Comment by fbarfi** [comment]
I just became aware that Giotto has created a giotto-deep (TDA for deep learning) which uses PyTorch instead of Tensorflow. Giotto-deep works seamlessly in using the GPU. Hence the above question.

**Comment by mglisse** [comment]
Hello,
as far as I know, there does not exist code to compute persistent homology purely on the GPU, all codes use the CPU for a key part, so rips_layer could not be pure GPU.
Perslay, on the other hand, is a variant of deep-sets, so should be able to work on the GPU. For instance, if you have pre-computed diagrams, the whole Perslay should be able to run on the GPU. If that's not the case (I am not super familiar with tensorflow and that part of gudhi), we should do something about it.
Could you give more details about the kind of seamless integration that you are missing in gudhi?

**Comment by mglisse** [comment]
I am going to transfer this issue to gudhi-devel (the appropriate repository, gudhi-deploy is just internal for CI stuff). I am posting this so you are not surprised and can find it back.

**Comment by fbarfi** [comment]
tTank you for your prompt response. What I am trying to do is optimize the persistence diagrams and persistence images at the same time (in the same loop calculating the gradients). It works fine for small datasets and it gives amazing results as to the improvements for the persistence images ( when compared to computing the persistence images after optimizing the persistence diagrams and storing them in files). However, for larger datasets it takes a very long time and is not practical anymore. I have an Apple Silicon M2 and I always try to use the GPU in combining TDA and machine learning. So far Gudhi has allowed me to achieve amazing results.
